### PR TITLE
Add `InputDatePickerFormField.focusNode` prop

### LINKED ISSUE #105881: [Proposal] Add `FocusNode` to `InputDatePickerFormField`
Hello

The `InputDatePickerFormField` widget must contain a `FocusNode`  for easy selected from keyboard.

Thanks

### PR DESCRIPTION
**Pull request by piedcipher**
Adds `focusNode` prop to `InputDatePickerFormField` widget.

Fixes #105881 

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Review by gspencergoog — APPROVED** [review]
32384589-a60f0e74-c078-11e7-9bc1-e5b5287aea9d

Good idea, thanks for the PR!

File: packages/flutter/test/material/input_date_picker_form_field_test.dart [hunk 1]
```diff
@@ -378,4 +380,18 @@ void main() {
       expect(find.text(errorFormatText), findsOneWidget);
     });
   });
+
+  testWidgets('FocusNode can request focus', (WidgetTester tester) async {
+    final FocusNode focusNode = FocusNode();
+    await tester.pumpWidget(inputDatePickerField(
+      focusNode: focusNode,
+    ));
+    expect(focusNode.hasFocus, isFalse);
```
Comment (chunhtai) [inline packages/flutter/test/material/input_date_picker_form_field_test.dart:389]: Can you add a check that the focusNode in TextFormField is the same as the one that passes in?

Comment (piedcipher) [reply packages/flutter/test/material/input_date_picker_form_field_test.dart:389]: Sure, will this test be part of TextFormField or InputDatePickerFormField?

Comment (piedcipher) [reply packages/flutter/test/material/input_date_picker_form_field_test.dart:390]: Done 👍✅, could you please check?

Comment (chunhtai) [reply packages/flutter/test/material/input_date_picker_form_field_test.dart:390]: This looks good, thanks!

**Review by chunhtai — APPROVED** [review]
LGTM

**Pull request merged**